#! /usr/bin/env python

PACKAGE='visnav2013_exercise3'
import roslib; roslib.load_manifest(PACKAGE)


from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("enable",              bool_t, 0,    "",                   False)
gen.add("c_prop_trans",              double_t, 0,    "const for P-part of translation",                   0.15,     0, 2)
gen.add("c_int_trans",              double_t, 0,    "const for I-part of translation",                   0,  0   , 2)
gen.add("c_deriv_trans",              double_t, 0,    "const for D-part of translation",                   0,     0, 2)

gen.add("c_prop_yaw",              double_t, 0,    "const for P-part of translation",                   0.15,     0, 2)
gen.add("c_int_yaw",              double_t, 0,    "const for I-part of translation",                   0,     0, 2)
gen.add("c_deriv_yaw",              double_t, 0,    "const for D-part of translation",                   0,     0, 2)


exit(gen.generate(PACKAGE, "visnav2013_exercise3", "PidParameter"))
